teuthology/orchestra/remote.py

```diff
@@ -300,6 +300,23 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
+    def is_mounted(self, path):
+        """
+        Check if the given path is mounted on the remote machine.
+
+        This method checks the contents of "/proc/self/mounts" instead of
+        using "mount" or "findmnt" command since these commands hang when a
+        CephFS client is blocked and its mount point on the remote machine
+        is left unhandled/unmounted.
+
+        :param path: path on remote host
+
+        :raises: :class:`RuntimeError`: unexpected error occurred
+        """
+        if not os.path.isdir(path):
```

Comment:
This runs on the test machine not the remote?

Reply:
Circling back: I don't really think you need this check at all.

Reply:
> This runs on the test machine not the remote?

In CephFS QA code, we write `self.mount_a.client_remote.read_file()`, so `teuthology.orchestra.remote.Remote` is being used.

Reply:
> Circling back: I don't really think you need this check at all.

Initially I thought the same, after giving [this review comment](https://github.com/ceph/teuthology/pull/1786#discussion_r950554743) some time, I thought extra error-check won't hurt.

Reply:
I doubt that. Try it!

Reply:
Doubt and try what? If you can be more specific, I'll know what to look out for. :)

If `path` holds the value `ext4`, without this check it might return True.

Reply:
> > This runs on the test machine not the remote?
> 
> In CephFS QA code, we write `self.mount_a.client_remote.read_file()`, so `teuthology.orchestra.remote.Remote` is being used.

This method is run for the Remote class but that doesn't change that `os.path.isdir` will be called from the testing machine. So it's looking at the file system mounted there and not on the remote.

Reply:
Thank you.

I was under the impression that you meant "check if path is dir" was unnecessary.